frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -153,38 +155,40 @@ impl<T: Config> Pallet<T> {
 					Err(MinerError::SolutionCallInvalid)
 				}
 			})
-			.or_else::<MinerError, _>(|_| {
-				// if not present or cache invalidated, regenerate
-				let (call, _) = Self::mine_checked_call()?;
-				save_solution(&call)?;
-				Ok(call)
+			.or_else::<MinerError, _>(|error| {
+				log!(debug, "restoring solution failed due to {:?}", error);
+				match error {
+					MinerError::Feasibility(_) | MinerError::NoStoredSolution => {
```

Comment:
I feel that it would make more sense to refactor this to avoid matching on this error type. And have a complete enum match instead. Ideally we want to match:
if no solution stored or solution is not feasible, then we do mine a new one.
If solution is stored and less good than on chain one, then we do nothing.

Also considering that `solution is stored and less good than on chain one` is an "error" is not very accurate to me.

But I think the current code is correct, so I'm ok.

Reply:
made it a bit more explicit, I think good enough for this PR.